### PR TITLE
CDS-6587 - Upgrade opsworks-ruby cookbook for cds-tools

### PR DESCRIPTION
**Pull request by mileslane**
### Docker Image
**[CDS-6587](https://projects.mckessonspecialtyhealth.com/jira/browse/CDS-6587)**

**Changes:**
1. This docker image runs Ubuntu 18.04.
2. node 10.24.0 (a CVE remediation update)
3. chefdk not included due to cookbooks tests now being run on a dedicated image running ubuntu 16.04
4. We now use the version of rubygems supplied by FullStaq ruby (3.0.3).
5. The latest version of bundle (2.2.14) which is also specified in the Gemfile.lock for cds-tools.

### CODE REVIEW

File: Dockerfile-cds-tools [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+# Java
+ENV JAVA_HOME=/jdk1.6.0_45
+ENV PATH="${JAVA_HOME}/bin:${PATH}"
+COPY ./src/jdk-6u45-linux-x64.bin ./jdk-6u45-linux-x64.bin
+RUN chmod a+x ./jdk-6u45-linux-x64.bin
+RUN ./jdk-6u45-linux-x64.bin
+RUN rm ./jdk-6u45-linux-x64.bin
+# Java test runner
+RUN apt-get update && apt-get install -y ant
```
Comment (olbrich) [inline Dockerfile-cds-tools:12]: We don't need java on the cds-tools image.
```suggestion
```

File: Dockerfile-cds-tools [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+# Ruby
+ENV RUBY_VERSION 2.6
+RUN echo 'deb https://apt.fullstaqruby.org ubuntu-18.04 main' > /etc/apt/sources.list.d/fullstaq-ruby.list
+RUN curl -SLfO https://raw.githubusercontent.com/fullstaq-labs/fullstaq-ruby-server-edition/main/fullstaq-ruby.asc \
+  && apt-key add fullstaq-ruby.asc \
+  && apt-get update \
+  && apt-get install -y fullstaq-ruby-common fullstaq-ruby-$RUBY_VERSION
+
+ENV NODE_VERSION 10.24.0
+RUN curl -sL https://deb.nodesource.com/setup_10.x| bash - \
```
Comment (olbrich) [inline Dockerfile-cds-tools:62]: ❓ Does this script respect the NODE_VERSION or does it just install the latest LTS in that series?

File: Dockerfile-cds-tools [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+RUN curl -sL https://dl.yarnpkg.com/debian/pubkey.gpg | apt-key add - \
+  && echo "deb https://dl.yarnpkg.com/debian/ stable main" > /etc/apt/sources.list.d/yarn.list \
+  && apt-get update \
+  && apt-get install yarn=$YARN_VERSION-1
+
+# CodeClimate
+RUN curl --compressed -L https://codeclimate.com/downloads/test-reporter/test-reporter-latest-linux-amd64 > /usr/local/bin/cc-test-reporter \
+  && chmod a+x /usr/local/bin/cc-test-reporter
+
+ENV PATH /usr/lib/fullstaq-ruby/versions/2.6/bin/:$PATH
```
Comment (olbrich) [inline Dockerfile-cds-tools:76]: Put this right after the Ruby install

File: Dockerfile-cds-tools (hunk 2 above)
Comment (mileslane) [reply Dockerfile-cds-tools:53]: I will test locally, but I think it works.

Comment (mileslane) [reply Dockerfile-cds-tools:53]: Found an alternative method that worked when I tested an installing an older node.

**Comment by mileslane** [comment]
Please re-review.

**Review by olbrich — APPROVED** [review]

**Pull request merged**